sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/ReportControlBlock.java

```diff
@@ -38,74 +40,103 @@
+        this.name = name;
+        this.dataSetRef = dataSetRef;
+        this.id = id;
     }
 
     /**
-     * Gets Class type value for ReportControlBlock
-     * @return ReportControlBlock.class
+     * Create a new TTrgOps with default values
+     * @return new instance of TTrgOps
      */
-    @Override
-    protected Class<ReportControlBlock> getClassType() {
-        return ReportControlBlock.class;
+    private static TTrgOps newDefaultTrgOps() {
+        TTrgOps newDefaultTrgOps = new TTrgOps();
+        newDefaultTrgOps.setDchg(true);
+        newDefaultTrgOps.setQchg(true);
+        newDefaultTrgOps.setDupd(false);
+        newDefaultTrgOps.setPeriod(true);
+        newDefaultTrgOps.setGi(true);
+        return newDefaultTrgOps;
     }
 
     /**
-     * Get Service Type value for ReportControlBlock
-     * @return Report
+     * Create a new TReportControl.OptFields with default values
+     * @return new instance of TReportControl.OptFields
      */
-    @Override
-    public TServiceType getServiceType() {
-        return TServiceType.REPORT;
+    private static TReportControl.OptFields newDefaultOptFields() {
+        TReportControl.OptFields newDefaultOptFields = new TReportControl.OptFields();
+        newDefaultOptFields.setSeqNum(false);
+        newDefaultOptFields.setTimeStamp(false);
+        newDefaultOptFields.setDataSet(false);
+        newDefaultOptFields.setReasonCode(false);
+        newDefaultOptFields.setDataRef(false);
+        newDefaultOptFields.setEntryID(false);
+        newDefaultOptFields.setConfigRef(false);
+        newDefaultOptFields.setBufOvfl(true);
+        return newDefaultOptFields;
     }
 
     /**
-     * Gets ConfRev value for ReportControlBlock
-     * @return 1L
+     * Constructor
+     * @param reportControl input
      */
+    public ReportControlBlock(TReportControl reportControl) {
+        super();
+        id = reportControl.getRptID();
+        desc = reportControl.getDesc();
+        name = reportControl.getName();
+        if(reportControl.isSetConfRev()) {
+            confRev = reportControl.getConfRev();
+        }
+        dataSetRef = reportControl.getDatSet();
+        indexed = reportControl.isIndexed();
+        intgPd = reportControl.getIntgPd();
+        buffered = reportControl.isBuffered();
+        bufTime = reportControl.getBufTime();
+        optFields = copyOptFields(reportControl.getOptFields());
+        trgOps = copyTTrgOps(reportControl.getTrgOps());
+        if (reportControl.isSetRptEnabled()){
+            rptEnabledMax = reportControl.getRptEnabled().getMax();
+            rptEnabledDesc = reportControl.getRptEnabled().getDesc();
+            targets = reportControl.getRptEnabled().getClientLN().stream().map(ControlBlockTarget::from).collect(Collectors.toCollection(ArrayList::new));
+        }
+    }
+
     @Override
-    protected Long getConfRev() {
-        return 1L;
+    public void setSecurityEnable(TPredefinedTypeOfSecurityEnum securityEnable) {
```

Comment:
To demonstrate why overloading this setter is required, I think it should be tested with the `ReportControlBlock`type.

Reply:
Setter was removed : see https://github.com/com-pas/compas-sct/pull/219#discussion_r1085149257